src/javascript/Edit/Edit.jsx

```diff
@@ -39,13 +46,31 @@ export const EditCmp = ({
                 sections,
                 values
             },
-            editCallback: nodeUuid => {
+            editCallback: (node, mutateNode) => {
```

Comment:
Rename handling should be done before the editCallback (because the edit callback is handling the close, so it will be based on history and we need to update the history before that)